src/client/Microsoft.Identity.Client/Platforms/iOS/iOSTokenCacheAccessor.cs

```diff
@@ -147,6 +146,25 @@ public IEnumerable<MsalAccountCacheItem> GetAllAccounts()
                 .ToList();
         }
 
+        internal string GetBrokerApplicationToken(string clientId)
+        {
+            var queryRecord = new SecRecord(SecKind.GenericPassword)
+            {
+                AccessGroup = _keychainGroup,
+                Account = clientId,
+                Service = iOSBrokerConstants.iOSBroker,
+            };
+
+            SecRecord record = SecKeyChain.QueryAsRecord(queryRecord, out SecStatusCode resultCode);
+
+            if (resultCode == SecStatusCode.Success)
+            {
+                return record.ValueData.ToString(NSStringEncoding.UTF8);
+            }
+
+            return null;
```

Comment:
Consider using `SecRecord TryGetBrokerApplicationToken(string clientId, out string appToken)` to be able to log SecRecord

Reply:
i think it makes more sense to return the secstatuscode, then the record, which may or may not have been saved to the keychain. at least this way, we will know if it's not a success, what the error is.

---
In reply to: [339124688](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1475#discussion_r339124688) [](ancestors = 339124688)